The job **`aarch64-gnu-llvm-21-1`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/26181734146/job/77026531086) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/77026531086) [(plain)](https://github.com/rust-lang/rust/commit/7ac006d2867c49f2fe57d0082a6522ab616b2c5f/checks/77026531086/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
To only update this specific test, also pass `--test-args reborrow/reborrow_multi_field_validation.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/reborrow/reborrow_multi_field_validation.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2" "--target=aarch64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/reborrow/reborrow_multi_field_validation" "-A" "unused" "-W" "unused_attributes" "-A" "internal_features" "-A" "incomplete_features" "-A" "unused_parens" "-A" "unused_braces" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/aarch64-unknown-linux-gnu/native/rust-test-helpers"
stdout: none
--- stderr -------------------------------
error[E0277]: the trait bound `String: Copy` is not satisfied
##[error]  --> /checkout/tests/ui/reborrow/reborrow_multi_field_validation.rs:10:5
   |
LL |     second: String, //~ ERROR the trait bound `String: Copy` is not satisfied
   |     ^^^^^^^^^^^^^^ the trait `Copy` is not implemented for `String`

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0277`.
```

</details>

--- a/compiler/rustc_hir_analysis/src/coherence/builtin.rs
+++ b/compiler/rustc_hir_analysis/src/coherence/builtin.rs
@@ -563,8 +563,8 @@ pub(crate) fn reborrow_info<'tcx>(
         )
         .is_ok()
         {
-            // Field implements Reborrow.
-            return Ok(());
+            // Field implements Reborrow, check remaining fields.
+            continue;
         }
 
         // Field does not implement Reborrow: it must be Copy.

--- a/tests/ui/reborrow/reborrow_multi_field_validation.rs
+++ b/tests/ui/reborrow/reborrow_multi_field_validation.rs
@@ -1,0 +1,15 @@
+#![feature(reborrow)]
+
+use std::marker::Reborrow;
+
+// Regression test: `reborrow_info` must validate ALL data fields,
+// not just stop at the first Reborrow field.
+
+struct Bad<'a> {
+    first: &'a mut i32,
+    second: String, //~ ERROR the trait bound `String: Copy` is not satisfied
+}
+
+impl<'a> Reborrow for Bad<'a> {}
+
+fn main() {}

--- a/tests/ui/reborrow/reborrow_multi_field_validation.stderr
+++ b/tests/ui/reborrow/reborrow_multi_field_validation.stderr
@@ -1,0 +1,9 @@
+error[E0277]: the trait bound `String: Copy` is not satisfied
+  --> $DIR/reborrow_multi_field_validation.rs:10:5
+   |
+LL |     second: String,
+   |     ^^^^^^^^^^^^^^ the trait `Copy` is not implemented for `String`
+
+error: aborting due to 1 previous error
+
+For more information about this error, try `rustc --explain E0277`.

